holoviews/core/data/grid.py

```diff
@@ -600,14 +615,20 @@ def range(cls, dataset, dimension):
             expanded = cls.irregular(dataset, dimension)
             column = cls.coords(dataset, dimension, expanded=expanded, edges=True)
         else:
-            column = dataset.dimension_values(dimension)
+            column = cls.values(dataset, dimension, flat=False)
         if column.dtype.kind == 'M':
-            return column.min(), column.max()
+            dmin, dmax = column.min(), column.max()
+            if da and isinstance(column, da.Array):
+                return dmin.compute(), dmax.compute()
+            return dmin, dmax
         elif len(column) == 0:
             return np.NaN, np.NaN
         else:
             try:
-                return (np.nanmin(column), np.nanmax(column))
+                dmin, dmax = (np.nanmin(column), np.nanmax(column))
+                if da and isinstance(column, da.Array):
+                    return dmin.compute(), dmax.compute()
```

Comment:
Would use `da.compute(dmin, dmax)` here as well. Reasoning is the same as above.

Reply:
Thanks, I keep forgetting about this!